consul/rtt.go

```diff
@@ -396,3 +379,36 @@ func getDatacenterMaps(s serfer, dcs []string) []structs.DatacenterMap {
 	}
 	return maps
 }
+
+// TODO (slackpad) - Add a unit test for this!
+
+// getDatacentersByDistance will return the list of DCs, sorted in order
+// of increasing distance based on the median distance to that DC from all
+// servers we know about in the WAN gossip pool. This will sort by name all
+// other things being equal (or if coordinates are disabled).
+func (s *Server) getDatacentersByDistance() ([]string, error) {
+	s.remoteLock.RLock()
+	defer s.remoteLock.RUnlock()
+
+	var dcs []string
+	for dc := range s.remoteConsuls {
```

Comment:
This one includes the local DC but it's called by a function further up that filters the local DC out for query purposes.